src/toolbar.scss

```diff
@@ -0,0 +1,170 @@
+$toolbar-height-cozy: 2.75rem;
+
+$toolbar-separator-height-compact: 1.5rem;
+$toolbar-separator-height-cozy: 2rem;
+
+$toolbar-spacing: 0.5rem;
+$toolbar-padding: 0 0.5rem;
+$title-toolbar-padding: 0 0.5rem 0 0.1rem;
+$title-toolbar-height: 2.75rem;
+
+%toolbarStandardColors {
+  @include fd-hover() {
+    background-color: var(--sapList_Hover_Background);
+  }
+
+  @include fd-active() {
+    background-color: var(--sapActiveColor);
+  }
+}
+
+@mixin activeStyles() {
+  &.#{$block}--active {
+    @content;
+  }
+}
+
+@mixin toolbarAuto() {
+  @include activeStyles() {
+    @extend %toolbarStandardColors;
+  }
+}
+
+@mixin toolbarSolid() {
+  background-color: var(--sapBackgroundColor);
+
+  @include activeStyles() {
+    @extend %toolbarStandardColors;
+  }
+}
+
+@mixin toolbarTransparent() {
+  background-color: var(--sapToolbar_Background);
+
+  @include activeStyles() {
+    @extend %toolbarStandardColors;
+  }
+}
+
+@mixin toolbarInfo() {
+  // no contrast function color: contrast(@sapUiInfobarBackground, @ sapUiBaseText , @ sapUiContentContrastTextColor , @sapUiContentContrastTextThreshold);
+  $active-color: white;
+
+  // no contrast function: background: darken(@sapUiListBackground, 10);
+  background-color: var(--sapBackgroundColor);
+
+  @include activeStyles() {
+    background-color: var(--sapInfobar_Background);
+    color: $active-color;
+
+    @include fd-hover() {
+      background-color: var(--sapInfobar_Hover_Background);
+    }
+
+    @include fd-active() {
+      background-color: var(--sapInfobar_Active_Background);
+    }
+  }
+}
+
+@mixin toolbarStandard() {
+  border-bottom: solid 0.0625rem var(--sapGroup_TitleBorderColor);
+}
+
+@mixin titleToolbar() {
+  padding: $title-toolbar-padding;
+  height: $title-toolbar-height;
+}
+
+@mixin toolbarClear() {
+  border-bottom: none;
+}
+
+%spaerMargin {
+  margin-left: $toolbar-spacing;
+
+  @include fd-rtl() {
+    margin-left: 0;
+    margin-right: $toolbar-spacing;
+  }
+}
+
+@mixin toolbarSpacing() {
+  > * ~ *:not(.#{$block}__spacer) {
+    @extend %spaerMargin;
+  }
+
+  > .#{$block}__spacer + *:not(.#{$block}__spacer) {
+    margin-left: 0;
+    margin-right: 0;
+  }
+}
+
+.#{$block} {
+  height: $toolbar-height-compact;
+  padding: $toolbar-padding;
+  display: flex;
+  align-items: center;
+
+  @include toolbarSpacing();
+
+  &--title {
+    @include titleToolbar();
+  }
+
+  &--standard {
+    @include toolbarStandard();
+  }
+
+  &--clear {
+    @include toolbarClear();
+  }
+
+  &--auto {
+    @include toolbarAuto();
+  }
+
+  &--solid {
+    @include toolbarSolid();
+  }
+
+  &--transparent {
+    @include toolbarTransparent();
+  }
+
+  &--info {
+    @include toolbarInfo();
+  }
+
+  &--cozy {
+    height: $toolbar-height-cozy;
+    .#{$block}__separator {
+      height: $toolbar-separator-height-cozy;
+    }
+  }
+
+  &__spacer {
+    height: 100%;
+
+    &--auto {
+      flex: 1 auto;
+    }
+
+    &--fixed {
+      display: inline-block;
+    }
```

Comment:
One of these should be the default one and the other should overwrite the behaviour. The less modifier classes the better :) I would put the auto behaviour as default and fixed as a modifier class (like it's now)

Reply:
fixed